annotations/src/main/java/org/corfudb/annotations/TransactionalMethod.java

```diff
@@ -0,0 +1,25 @@
+package org.corfudb.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/** Marks that this method should be execute transactionally.
+ * Created by mwei on 3/29/16.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+public @interface TransactionalMethod {
```

Comment:
SMRMap does not contain other objects, so it does not need to do anything transactionally. FGMap however, does contain objects, and it sometimes needs to modify those objects transactionally.